### DISCUSSION TITLE
Porting from Atom - problem with init.coffee

### DISCUSSION THREAD

**Discussion by silarri** [pulsar-edit/.github#156] [Pulsar Support] [answered]
I'm trying to move all my configuration from Atom to Pulsar, but I am having some problems. Particularly, the contents of the init.coffee file seems to be ignored, as I define there some new commands and they have not been added. Any idea about why this may be happening?

I understand that CoffeScript is still supported, but JavaScript is somehow preferred? If so, is there a way to automatically translate the init.coffee file to the new JavaScript syntax?

I read what is said at https://github.com/orgs/pulsar-edit/discussions/63 about the automatic conversion, but if I understand it well this is something planned but not yet available. Is this correct? Should I wait before migrating?

Thank you for any tips.

**Comment by confused-Techie** [comment]
So on this, I don't beleive there's any logic yet to automatically move your configuration files. And while JavaScript is preferred, that really means the developer team prefers to program in JavaScript, and I don't see us removing support for CSON config files anytime soon.

But to the issue at hand, the biggest question is where is your `init.coffee` keep in mind Pulsar only looks for it's config files in `<home>/.pulsar/` so if you are making changes to your `init.coffee` in the old `.atom` directory then Pulsar won't see it.

And to the last point on waiting, I don't think it's all that needed. The most you need to do, in reference to config files, is install pulsar, then copy the entire `.atom` directory and place it in `.pulsar` theres very few changes as of yet, so it should mostly be fine. (Although you could skip moving `.atom/packages` if you want to be extra safe.

**Comment by silarri** [comment]
I did as suggested (I basically copied the whole .atom folder as .pulsar). My init.coffee file is in the <home>/.pulsar folder. However, I define several commands in the init.coffee file and then these commands are not available in Pulsar. For example, my file starts with:

atom.commands.add 'atom-text-editor', 'custom:myFirstCommand', ->
...

But then the command custom:myFirstCommand is not available in the Pulsar editor. This works well with Atom. Any idea of what may be happening?

**Reply by confused-Techie** [reply]
I'm not to sure, as far as I know we haven't made many, if any, changes to the command registry, @pulsar-edit/core any ideas on this one?

**Reply by Spiker985** [reply]
Nope, as far as I'm aware, no changes to command registration have been made.

The addition of command activation within snippets is the closest thing, but would have no bearing on this issue

**Reply by savetheclocktower** [reply]
I can think of a couple of things to try:

* Make sure there isn't an `init.js` in the same directory, just to rule out the idea that Pulsar is ignoring your `init.coffee` altogether.
* Open the dev tools (**View → Developer → Toggle Developer Tools**) and look in the console for any errors. If you don't see any, try adding a logging statement to the top of your `init.coffee`...

    ```coffeescript
    console.log("SANITY CHECK");
    ```

    …then reload the window (**View → Developer → Reload Window**), open the dev tools again after the reload, and see if your logged statement is present in the console.

**Reply by silarri** [reply]
Thanks a lot for all the suggestions and comments! This was it, I had a init.js file (the default one once Pulsar is installed) in the folder and so the init.coffee file was completely ignored. I removed the init.js file and now it seems to work well. Thank you very much.

**Comment by Daeraxa** [comment]
Could you maybe share your init.coffee to see if it can be replicated?

**Comment by confused-Techie** [comment, marked as answer]
To have a comment marked as an answer, that can be easily found again later.

As @savetheclocktower suggested the `.pulsar` folder contained a `init.js` (as it was created by default on installation) and was choosing that file over @silarri's `init.coffee`.

So the issue was resolved by removing the `init.js` file from the config folder.
